dbus-wait.sh

```diff
@@ -8,7 +8,10 @@ wait_for_dbus() {
 			  /org/freedesktop/DBus \
 			  org.freedesktop.DBus.ListNames
 
-		if [ $? ]; then
+		dbus_test="$?"
+                if $dbus_test | grep -q 'Failed to connect to socket'; then
```

Comment:
Do you think that `dbus-send` will sometimes return a 0 exit code when the message contains 'Failed to connect to socket'?

Also not clear if you are trying to pipe (`|`) or use or (`||`).

Reply:
Was trying to pipe to grep yes. To check for the string.

Basically the issue was that when showing "failed to connect to socket" it was still proceeding using the existing code, as shown in the linked comment above:

```
08.10.21 22:01:00 (+0200) helium-miner Failed to open connection to "system" message bus: Failed to connect to socket /host/run/dbus/session_bus_socket: No such file or directory
08.10.21 22:01:00 (+0200) helium-miner DBus is now accepting connections
```

So the idea was to have it so that if it says "can't connect to socket" or returns no output, then it will sleep. Otherwise it will continue.

Reply:
tl;dr: ya I think adding `-ne 0` is sufficient.

My understanding is that "Failed to open connection..." is just a message that is outputted to [stderr](https://gitlab.freedesktop.org/dbus/dbus/-/blob/dbus-1.12/tools/dbus-send.c#L406) during program execution. This is an orthogonal concern to the [numeric exit code](https://www.gnu.org/software/bash/manual/html_node/Exit-Status.html) returned by every linux process, where only 0 marks success. This error message corresponds to an [error code of 1](https://gitlab.freedesktop.org/dbus/dbus/-/blob/dbus-1.12/tools/dbus-send.c#L411).

The previous code was always evaluating to true. `if [ $? ]` is the same as ` if test $?`, which always evaluated to `if test 1` or `if test 2`. Both of those will always evaluate to true. Here's how I proved it to myself.

```
➜  test 1 -gt 2 && echo "True" || echo "False"
False
➜  test 0 && echo "True" || echo "False"
True
➜  test 1 && echo "True" || echo "False"
True
```

Reply:
Clarification: The way the code was organized _originally_, we actually want. `if [ $? -eq 0]`. 

```
                if [ $? -eq 0 ]; then
			break;
		else
			sleep 0.1
```